tests/Resources/modules/ps_banner/ps_banner.php

```diff
@@ -0,0 +1,265 @@
+* obtain it through the world-wide-web, please send an email
+* to license@prestashop.com so we can send you a copy immediately.
+*
+* DISCLAIMER
+*
+* Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+* versions in the future. If you wish to customize PrestaShop for your
+* needs please refer to http://www.prestashop.com for more information.
+*
+*  @author PrestaShop SA <contact@prestashop.com>
+*  @copyright  2007-2015 PrestaShop SA
+*  @license    http://opensource.org/licenses/afl-3.0.php  Academic Free License (AFL 3.0)
+*  International Registered Trademark & Property of PrestaShop SA
+*/
+
+if (!defined('_PS_VERSION_')) {
+    exit;
+}
+
+use PrestaShop\PrestaShop\Core\Module\WidgetInterface;
+
+class Ps_Banner extends Module implements WidgetInterface
+{
+    private $templateFile;
+
+	public function __construct()
+	{
+		$this->name = 'ps_banner';
+		$this->version = '2.1.0';
+		$this->author = 'PrestaShop';
+		$this->need_instance = 0;
+
+        $this->bootstrap = true;
+        parent::__construct();
+
+        $this->displayName = $this->trans('Banner', array(), 'Modules.Banner.Admin');
+        $this->description = $this->trans('Displays a banner on your shop.', array(), 'Modules.Banner.Admin');
+
+        $this->ps_versions_compliancy = array('min' => '1.7.1.0', 'max' => _PS_VERSION_);
+
+        $this->templateFile = 'module:ps_banner/ps_banner.tpl';
+    }
+
+    public function install()
+    {
+        return (parent::install() &&
+            $this->registerHook('displayHome') &&
+            $this->registerHook('actionObjectLanguageAddAfter') &&
+            $this->installFixtures() &&
+            $this->disableDevice(Context::DEVICE_MOBILE));
+    }
+
+    public function hookActionObjectLanguageAddAfter($params)
+    {
+        return $this->installFixture((int)$params['object']->id, Configuration::get('BANNER_IMG', (int)Configuration::get('PS_LANG_DEFAULT')));
+    }
+
+    protected function installFixtures()
+    {
+        $languages = Language::getLanguages(false);
+
+        foreach ($languages as $lang) {
+            $this->installFixture((int)$lang['id_lang'], 'sale70.png');
+        }
+
+        return true;
+    }
+
+    protected function installFixture($id_lang, $image = null)
+    {
+        $values['BANNER_IMG'][(int)$id_lang] = $image;
+        $values['BANNER_LINK'][(int)$id_lang] = '';
+        $values['BANNER_DESC'][(int)$id_lang] = '';
+
+        Configuration::updateValue('BANNER_IMG', $values['BANNER_IMG']);
+        Configuration::updateValue('BANNER_LINK', $values['BANNER_LINK']);
+        Configuration::updateValue('BANNER_DESC', $values['BANNER_DESC']);
+    }
+
+    public function uninstall()
+    {
+        Configuration::deleteByName('BANNER_IMG');
+        Configuration::deleteByName('BANNER_LINK');
+        Configuration::deleteByName('BANNER_DESC');
+
+        return parent::uninstall();
+    }
+
+    public function postProcess()
```

Comment:
You can remove many lines of this module, more because it's only for test.

Reply:
You're right but it's not like it changes a lot anyway. Besides when the legacy tests will be migrated this could be useful especially the `registerHook` part so I prefer to leave it like this for now.